library/spdm_common_lib/libspdm_com_crypto_service.c

```diff
@@ -205,14 +205,18 @@ boolean spdm_calculate_m1m2(IN void *context, IN boolean is_mut,
             return FALSE;
         }
 
-        // debug only
-        spdm_hash_all(
-            spdm_context->connection_info.algorithm.base_hash_algo,
-            get_managed_buffer(&m1m2),
-            get_managed_buffer_size(&m1m2), hash_data);
-        DEBUG((DEBUG_INFO, "m1m2 Mut hash - "));
-        internal_dump_data(hash_data, hash_size);
-        DEBUG((DEBUG_INFO, "\n"));
+        // Debug code only - calculate and print value of m1m2 mut hash
+        DEBUG_CODE(
+            if (!spdm_hash_all(
+                spdm_context->connection_info.algorithm.base_hash_algo,
+                get_managed_buffer(&m1m2),
+                get_managed_buffer_size(&m1m2), hash_data)) {
+                return FALSE;
+            }
+            DEBUG((DEBUG_INFO, "m1m2 Mut hash - "));
+            internal_dump_data(hash_data, hash_size);
+            DEBUG((DEBUG_INFO, "\n"));
```

Comment:
I guess these don't need to be inside DEBUG_CODE. But it looks a little cleaner all in one, especially since these function calls rely on the result of the former.